src/mui/input/DateInput.js

```diff
@@ -31,6 +32,7 @@ class DateInput extends Component {
             autoOk
             value={datify(input.value)}
             onChange={this.onChange}
+            onBlur={this.onBlur}
```

Comment:
The purpose is to run `input.onBlur()` rather than `input.onBlur(event)` when DateInput send `onBlur` event. For more info, see here erikras/redux-form#1218.

I am not sure how to write a unit test because the problem happens when user clicks on the input.
The input sends the event to redux-form and redux-form update its form value to the string value on the input (something like `01/02/2017`). 
Then DateInput updates its value to `datify("01/02/2017")`.

So for locale like `pt`, month and date will be swapped when user clicks on the input.

Maybe e2e test is more suitable?